draft-ietf-jsonpath-base.md

```diff
@@ -1535,36 +1523,80 @@ A function expression is well-typed if all of the following are true:
-| empty nodelist         | `Nothing`             | `LogicalFalse`       |
-| single-node nodelist   | the node's JSON value | `LogicalTrue`        |
-| multiple-node nodelist | `Nothing`             | `LogicalTrue`        |
-{: #tbl-typeconv title="Conversion of nodelists to values and logicals"}
+Note that a multiple-node nodelist result is deemed `LogicalTrue` in
+the test expression, but cannot be used for obtaining a single
+value and thus feeds the comparison expression with `Nothing`.
 
-For example, given a function `myfunc()` of declared result type
-`NodesType`, the following filter expressions are both valid but can
-have different and unexpected results:
 
-| function result        | `myfunc(@.a)` evaluation (as a test expression)                              | `myfunc(@.a)==42` evaluation (in a comparison)                                |
-| :-:                    | :-                                                                           | :-                                                                            |
-| empty nodelist         | The nodelist is converted to `LogicalFalse`.<br>The expression result is false. | The empty nodelist is converted to `Nothing`.<br>The expression result is false. |
-| single-node nodelist   | The nodelist is converted to `LogicalTrue`.<br>The expression result is true.   | The nodelist is converted to the node's value.<br>The expression result is true. |
-| multiple-node nodelist | The nodelist is converted to `LogicalTrue`.<br>The expression result is true.   | The nodelist is converted to `Nothing`.<br>The expression result is false.       |
+### `exist` Function Extension {#exist-fn}
 
-Note that a multiple-node nodelist result is deemed `LogicalTrue` in
-the test expression, but cannot be used for conversion to a single
-value and thus feeds a comparison with `Nothing`.
+Parameters:
+: 1. `NodesType`
+
+Result:
+: `LogicalType`
+
+The "exist" function extension returns `LogicalTrue` if the nodelist
+given consists of at least one node, `LogicalFalse` otherwise:
+
+~~~ JSONPath
+$[?exist(fn(@.*.price))]
+~~~
+
+The effect of the `exist` function is similar to the existence test
+that is performed when a query expression is directly used in a test
+expression:
+If the nodelist contains one or more nodes, the result is
+`LogicalTrue`.
+If the nodelist is empty, the result is `LogicalFalse`.
+
+
+### `value` Function Extension {#value-fn}
+
+Parameters:
+: 1. `NodesType`
+
+Result:
+: `ValueType`
+
+The "value" function extension provides a way to obtain the value of a
+node in a single-element nodelist and make that available for further
+processing in the filter expression:
+
+~~~ JSONPath
+$[?value(fn(@.*.price)) >= 7.99]
+~~~
+
+Its only argument is a nodelist.
+If the nodelist contains a single node, the result is the value of the node.
+If the nodelist is empty or contains multiple nodes, the result is `Nothing`.
```

Comment:
I'm not sure converting multiple nodes to `Nothing` is the best approach. I would prefer to have a type for single node nodelists. Then `value()` could take a parameter of that type and attempting to pass a filter-path` to `value()` would not be well-typed.

Reply:
See https://github.com/ietf-wg-jsonpath/draft-ietf-jsonpath-base/pull/410#issuecomment-1445311058 for my view of this.